.github/workflows/on-pull-request.yml

```diff
@@ -0,0 +1,95 @@
+---
+# Full PR suite.  PRs need to pass this pipeline in order to be moved to the
+# merge queue, where they'll use the next rule as the branch-protection rule
+# for the final merge to the base branch.
+name: Pull Requests
+on:
+  pull_request:
+    branches: [main, "stable/*"]
+  merge_group:
```

Comment:
Is this is needed?

Reply:
I don't think so, this workflow should only run during pull-requests, the one for the merge queue is different and defined in `on-merge-queue`

Reply:
Removed in [eb9cf50](https://github.com/Qiskit/qiskit/pull/13474/commits/eb9cf5070b1afc7d6917d4b9bf1cb08a2a23d940)